input/input.go

```diff
@@ -44,31 +46,59 @@ func NewDefaultHandler(metrics mdata.Metrics, metricIndex idx.MetricIndex, input
 	}
 }
 
-// process makes sure the data is stored and the metadata is in the index
+// ProcessMetricPoint updates the index if possible, and stores the data if we have an index entry
 // concurrency-safe.
-func (in DefaultHandler) Process(metric *schema.MetricData, partition int32) {
-	if metric == nil {
+func (in DefaultHandler) ProcessMetricPoint(point schema.MetricPoint, partition int32) {
+	in.metricsReceived.Inc()
+	if !point.Valid() {
+		in.MetricInvalid.Inc()
+		log.Debug("in: Invalid metric %v", point)
+		return
+	}
+
+	pre := time.Now()
+	archive, _, ok := in.metricIndex.UpdateMaybe(point, partition)
+	in.pressureIdx.Add(int(time.Since(pre).Nanoseconds()))
+
+	if !ok {
```

Comment:
We should keep a metric of the number of points being dropped due to not existing in the index.